tests/model_fields/test_jsonfield.py

```diff
@@ -360,7 +360,7 @@ def test_ordering_grouping_by_count(self):
         qs = NullableJSONModel.objects.filter(
             value__isnull=False,
         ).values('value__d__0').annotate(count=Count('value__d__0')).order_by('count')
-        self.assertQuerysetEqual(qs, [1, 11], operator.itemgetter('count'))
+        self.assertQuerysetEqual(qs, [0, 1], operator.itemgetter('count'))
```

Comment:
I think this is actually the correct behaviour here.

Before this change, `Count('value__d__0')` was equivalent to `Count('value')`, because the `__d__0` transform/lookup was ignored.

Query before:

```sql
SELECT ("model_fields_nullablejsonmodel"."value" #> ['d', '0']),
        COUNT("model_fields_nullablejsonmodel"."value") AS "count"
        FROM "model_fields_nullablejsonmodel"
    WHERE "model_fields_nullablejsonmodel"."value" IS NOT NULL
    GROUP BY ("model_fields_nullablejsonmodel"."value" #> ['d', '0'])
    ORDER BY "count" ASC
```

Query after:

```sql
SELECT ("model_fields_nullablejsonmodel"."value" #> ['d', '0']),
        COUNT(("model_fields_nullablejsonmodel"."value" #> ['d', '0'])) AS "count"
        FROM "model_fields_nullablejsonmodel"
    WHERE "model_fields_nullablejsonmodel"."value" IS NOT NULL
    GROUP BY ("model_fields_nullablejsonmodel"."value" #> ['d', '0'])
    ORDER BY "count" ASC
```

The other possible change to this test would be to change the queryset to:

```python
        qs = NullableJSONModel.objects.filter(
            value__isnull=False,
        ).values('value__d__0').annotate(count=Count('value')).order_by('count')
```

Reply:
yeah you're right, the previous test was affected by the silent truncation of transforms from field references as reported by users. 

I think the original intent was to do `count=Count('value')` and not `Count('value__d__0')` though so I'd switch it to do the latter.

Reply:
This test was added in d87bd29c4f8dfcdf3f4a4eb8340e6770a2416fe3 to confirm fix for ticket-29139. The original intent was to do `Count('value__d__0')`. Please update it.